src/test/regress/sql/multi_citus_tools.sql

```diff
@@ -257,5 +254,73 @@ UPDATE pg_dist_shard_placement SET shardstate = 3 WHERE shardid % 2 = 0;
 SELECT * FROM run_command_on_shards('check_shards', 'select 1');
 DROP TABLE check_shards CASCADE;
 
+-- test the connections to worker nodes
+SELECT bool_and(success) AS all_nodes_are_successful FROM (
+    SELECT citus_check_connection_to_node(nodename, nodeport) AS success
+    FROM pg_dist_node
+    WHERE isactive = 't' AND noderole='primary'
+) subquery;
+
+-- verify that the coordinator can connect to itself
+SELECT citus_check_connection_to_node('localhost', :master_port);
+
+-- verify that the connections are not successful for wrong port
+-- test with invalid port, prevent OS dependent warning from being displayed
+SET client_min_messages TO ERROR;
+SELECT citus_check_connection_to_node('localhost', nodeport:=1234);
+
+-- verify that the connections are not successful due to timeouts
+SET citus.node_connection_timeout TO 10;
+SELECT citus_check_connection_to_node('www.citusdata.com');
+RESET citus.node_connection_timeout;
+
+SET client_min_messages TO DEBUG;
+
+-- check the connections in a transaction block
+BEGIN;
+SELECT citus_check_connection_to_node(nodename, nodeport)
+FROM pg_dist_node
+WHERE isactive = 't' AND noderole='primary';
+
+CREATE TABLE distributed(id int, data text);
+SELECT create_distributed_table('distributed', 'id');
+SELECT count(*) FROM distributed;
```

Comment:
minor:
```SQL
INSERT INTO distributed SELECT i, i::text FROM generate_series(0,100)i;
set LOCAL client_min_messages to error;
SELECT bool_and(citus_check_connection_to_node(data, id)) from distributed;
```
It's an interesting way to use `citus_check_connection_to_node` on the workers, via executor.

Reply:
I could not get this piece of test to work. Here are my failed attempt logs:

```
DEBUG:  Router planner cannot handle multi-shard select queries
ERROR:  canceling statement due to statement timeout
```